### DISCUSSION TITLE
Need help and guidance about software development

### DISCUSSION THREAD

**Discussion by Rushi-07** [github/india#804] [Students]
Hi.
I'm RUSHIKESH.  I have some doubts about software development. I'm confused I need help and Proper step by step guidance.
and I am new to git hub.

**Comment by ey-mentor1** [comment]
Hello, 
Can you please specify the issue that you are facing.
And regarding the GitHub hands-on, there is one separate module in the FRT portal.
Please check that too.
Thank you

**Reply by Rushi-07** [reply]
Hello,
my issue is i just know how to code in any IDE but from now on i need to know, how software are actually created. like i want to know which tools are being used for developing software or any application. i mean how front end and backend of the application work together. how they are connected. how the cloud and database is connected to our application. and what is framework what is API. these all things i want to learn. could you please guide me ?

Thank you.

**Reply by ey-mentor1** [reply]
Hello,
There is no such one way to learn entire thing.
I can only suggest you to look for programming language, both front-end and back-end. Get familiarise with the language, understand how both of them can be used together.
Then make some projects and post that learn to integrate the same project using cloud.

Regarding source material, please use the authentice documentation from the original sites of the programing language that you choose.
You can also refer YouTube, there are many good channels out there.

You have to do lot of research and then have to draw plan.

Happy leaning!!

Thank you
